### ISSUE TITLE
unwrap-used and expect-used don't trigger on unwrap_err/expect_err

### ISSUE THREAD

**Issue by sgued** [rust-lang/rust-clippy#9331]
### Summary

The `unwrap_used` and `expect_used` lints forbid the use of `unwrap` and `expect`. It seems to me that these lints should also lint `unwrap_err` and `expect_err` as they have essentially the same behavior. I believe people who want to not use unwrap to avoid panic probably don't want a panic to be caused by `unwrap_err` either.

### Reproducer

This does not produce any error related to `unwrap_used` and `expect_used`
```rust
#[deny(clippy::unwrap_used, clippy::expect_used)]
fn main() {
    let mut a = Ok(3);
    a = Err(3);
    a.unwrap_err();
    a = Err(3);
    a.expect_err("expect");
}
```
but i think it should produce errors similar to 
```rust
#[deny(clippy::unwrap_used, clippy::expect_used)]
fn main() {
    let mut a = Ok(3);
    a = Err(3);
    a.unwrap();
    a = Err(3);
    a.expect("expect");
}
```

### Version

```text
rustc 1.65.0-nightly (f22819bcc 2022-08-12)
binary: rustc
commit-hash: f22819bcce4abaff7d1246a56eec493418f9f4ee
commit-date: 2022-08-12
host: x86_64-unknown-linux-gnu
release: 1.65.0-nightly
LLVM version: 15.0.0
```

### Additional Labels

@rustbot label -C-bug +C-enhancement
